### ISSUE TITLE
jpgd::jpeg_decoder::check_sample_buf_ofs(int) const: Assertion `ofs >= 0' failed.

### ISSUE THREAD

**Issue by arpu** [KhronosGroup/KTX-Software#404]
Hey,

get this error on wolf3d Avatars (https://readyplayer.me/)
```
toktx --t2 --2d --genmipmap --nowarn --bcmp --clevel 1 --qlevel 192 --srgb -- /tmp/gltfpack-pufIhl.ktx2 /tmp/gltfpack-DCgM4j.jpg => 0
toktx --t2 --2d --genmipmap --nowarn --bcmp --clevel 1 --qlevel 192 --normal_map --linear -- /tmp/gltfpack-eEVzpk.ktx2 /tmp/gltfpack-m5JxEn.jpg => 0
toktx: /home/arpu/Work/githubsources/KTX-Software/lib/basisu/encoder/jpgd.h:275: int jpgd::jpeg_decoder::check_sample_buf_ofs(int) const: Assertion `ofs >= 0' failed.
toktx --t2 --2d --genmipmap --nowarn --resize 4x4 --bcmp --clevel 1 --qlevel 192 --srgb -- /tmp/gltfpack-tloSko.ktx2 /tmp/gltfpack-uq23Hk.jpg => 134
Warning: unable to encode image 2, skipping
```
using KTX-Software master and meshoptimizer master

**Comment by arpu** [comment]
the problematic file is body_texture.jpg

```
toktx --t2 --2d --genmipmap --nowarn --bcmp --clevel 1 --qlevel 192 --normal_map --linear /home/arpu/Downloads/wolf/boots_Gloss_01.ktx2 /home/arpu/Downloads/wolf/body_texture.jpg -vv
toktx: /home/arpu/Work/githubsources/KTX-Software/lib/basisu/encoder/jpgd.h:275: int jpgd::jpeg_decoder::check_sample_buf_ofs(int) const: Assertion `ofs >= 0' failed.
[1]    1691918 IOT instruction (core dumped)  toktx --t2 --2d --genmipmap --nowarn --bcmp --clevel 1 --qlevel 192  --linear
```

files attached
![body_texture](https://user-images.githubusercontent.com/14173/115915395-9270ea80-a473-11eb-99ba-6723a0e0f866.jpg)

**Comment by MarkCallow** [comment]
The attached file shows up as black in my web browser when I click the image. Is that correct? Since it is a 1x1 image its kind of hard to tell. It's barely visible in the issue comment.

**Comment by arpu** [comment]
yes this is correct

![Bildschirmfoto von 2021-04-25 10-34-08](https://user-images.githubusercontent.com/14173/115986702-ce11ce80-a5b1-11eb-8ec1-b8a22b063e65.png)

**Comment by arpu** [comment]
anything i can provide to fix this?

**Comment by arpu** [comment]
thx, @MarkCallow  with the Release Build i get a valid viewable glb now! 

![Bildschirmfoto von 2021-04-29 10-49-42](https://user-images.githubusercontent.com/14173/116525336-b3549800-a8d8-11eb-89b3-6353481c31f3.png)

**Comment by arpu** [comment]
this should be fixed with new basisu release 1.16
